src/components/main/index.tsx

```diff
@@ -17,36 +17,34 @@ import Loader from "../loader";
 type State = {
   movies: Movie[],
   moviesQuantity: number,
-  loading: boolean
+  reload: boolean,
+  loading: boolean,
+  typeSelected: string,
+  sortBySelected: string
 }
 
 function Main() {
   // Dispatch Instance
   const dispatch = useDispatch<AppDispatch>();
 
   // Redux State Extraction
-  const { movies, moviesQuantity, loading } = useSelector<RootState, State>((state) => state.movies);
-
-  // Local State
-  const [typeSelected, setTypeSelected] = React.useState("all");
-  const [sortBySelected, setSortBySelected] = React.useState("release_date");
+  const {
+    movies, moviesQuantity, reload, loading, typeSelected, sortBySelected,
+  } = useSelector<RootState, State>((state) => state.movies);
 
   React.useEffect(() => {
-    if (typeSelected === "all") {
-      dispatch(getAllMoviesAction("0", sortBySelected));
-    } else {
-      dispatch(getAllMoviesByGenreAction("0", typeSelected, sortBySelected));
+    if (reload) {
+      if (typeSelected === "all") {
```

Comment:
please do not use magic strings or numbers. Use constants instead.
The best approach here will be change `CATEGORIES` to enum i guess

Reply:
Resolved